docs/extensions.adoc

```diff
@@ -117,6 +117,7 @@ To make predicates easier to read and write, the following properties are availa
 * `env` A map of all environment variables
 * `os` Information about the operating system (see `spock.util.environment.OperatingSystem`)
 * `jvm` Information about the JVM (see `spock.util.environment.Jvm`)
+* `instance` The specification instance if instance fields, shared fields or instance methods are needed
```

Comment:
we should mention that instance will require the test to execute to that point, similar as with the data-driven variables.